### PR TITLE
ref(deps): Bump sentry-relay to v0.8.48

### PR DESCRIPTION
**Pull request by iker-barriocanal**
Along the way of bumping dependencies, some tests are fixed to cover for the changes in Relay.

### CODE REVIEW

File: src/sentry/eventstore/models.py [hunk 1]
```diff
@@ -210,7 +210,10 @@ def get_minimal_user(self) -> User:
                 {"id": user_id, "email": email, "username": username, "ip_address": ip_address}
             )
 
-        return self.get_interface("user")
+        user = self.get_interface("user")
+        if user is not None:
+            user._data.pop("sentry_user", None)
+            return user
```
Comment (iker-barriocanal) [inline src/sentry/eventstore/models.py:216]: Fixes the test `tests/sentry/eventstore/test_models.py::EventTest::test_snuba_data`.

`get_minimal_user` intends to extract the _minimal_ user that may be exposed to external services, so there's no reason to expose `sentry_user`. Alternatively, we could expose the value from Snuba, but I see no reason to do that as the field was originally added for DDM.

File: src/sentry/receivers/features.py [hunk 2]
```diff
@@ -122,7 +122,7 @@ def record_event_processed(project, event, **kwargs):
     # Check to make sure more the ip address is being sent.
     # testing for this in test_no_user_tracking_for_ip_address_only
     # list(d.keys()) pattern is to make this python3 safe
-    if user_context and list(user_context.keys()) != ["ip_address"]:
+    if user_context and list(user_context.keys()) != ["ip_address", "sentry_user"]:
```
Comment (iker-barriocanal) [inline src/sentry/receivers/features.py:125]: Fixes the test `test_no_user_tracking_for_ip_address_only`.

Relay now extracts `sentry_user` automatically from the IP address, and both should be filtered out together and not trigger user tracking.

File: tests/sentry/event_manager/test_normalization.py [hunk 3]
```diff
@@ -42,7 +42,7 @@ def test_interface_is_relabeled():
     manager.normalize()
     data = manager.get_data()
 
-    assert data["user"] == {"id": "1"}
+    assert data["user"] == {"id": "1", "sentry_user": "id:1"}
```
Comment (iker-barriocanal) [inline tests/sentry/event_manager/test_normalization.py:45]: `sentry_user` is now automatically added in Relay.

File: tests/sentry/event_manager/interfaces/snapshots/test_spans/test_multiple_full.pysnap [hunk 4]
```diff
@@ -1,13 +1,11 @@
 ---
-created: '2019-07-11T19:30:21.322813Z'
-creator: sentry
 source: tests/sentry/event_manager/interfaces/test_spans.py
 ---
 errors: null
 to_json:
 - data:
+    http.response.status_code: 200
```
Comment (iker-barriocanal) [inline tests/sentry/event_manager/interfaces/snapshots/test_spans/test_multiple_full.pysnap:7]: Relay renamed the value for this key. It seems it didn't have a negative impact in other places. (same for the snapshot below)

File: src/sentry/eventstore/models.py [hunk 5]
```diff
@@ -210,7 +210,11 @@ def get_minimal_user(self) -> User:
                 {"id": user_id, "email": email, "username": username, "ip_address": ip_address}
             )
 
-        return self.get_interface("user")
+        user = self.get_interface("user")
+        if user is not None:
```
Comment (Dav1dde) [inline src/sentry/eventstore/models.py:214]: As discussed, remove these changes and fix the test to only check id, email, username and ip

**Review by jjbayer — APPROVED** [review]
Pre-approving, please apply changes as discussed.

File: src/sentry/receivers/features.py (hunk 2 above)
Comment (Dav1dde) [inline src/sentry/receivers/features.py:125]: Update to a set comparison

**Review by Dav1dde — APPROVED** [review]

File: src/sentry/eventstore/models.py (hunk 5 above)
Comment (iker-barriocanal) [reply src/sentry/eventstore/models.py]: Done in [071238f](https://github.com/getsentry/sentry/pull/66401/commits/071238f3ab6bd78283e0d2bcac153ade30e9c30c).

File: src/sentry/receivers/features.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (iker-barriocanal) [reply src/sentry/receivers/features.py]: Done in [071238f](https://github.com/getsentry/sentry/pull/66401/commits/071238f3ab6bd78283e0d2bcac153ade30e9c30c).

File: src/sentry/eventstore/models.py (hunk 1 above)
Comment (iker-barriocanal) [reply src/sentry/eventstore/models.py]: Done in [071238f](https://github.com/getsentry/sentry/pull/66401/commits/071238f3ab6bd78283e0d2bcac153ade30e9c30c).

**Pull request merged**